pydra/utils/messenger.py

```diff
@@ -0,0 +1,111 @@
+import abc
+import datetime as dt
+import enum
+from pathlib import Path
+import os
+
+def gen_uuid():
+    import uuid
+    return uuid.uuid4().hex
+
+
+def now():
+    return dt.datetime.utcnow().isoformat(timespec='microseconds')
+
+
+class AuditFlag(enum.Flag):
+    NONE = 0
+    PROV = enum.auto()  # 0x01
+    RESOURCE = enum.auto()  # 0x02
+    ALL = PROV | RESOURCE
+
+
+class RuntimeHooks(enum.IntEnum):
```

Comment:
The only reason to use `IntEnum` over `Enum` is that `IntEnum` is comparable with `int`s outside the class. Is that the goal here?

See the note at the bottom of [enum.Flag](https://docs.python.org/3/library/enum.html#flag).